File: src/main/java/nl/hanze/week1/assignment1/observers/AAPLObserver.java
package nl.hanze.week1.assignment1.observers;

import nl.hanze.week1.assignment1.Observer;
import nl.hanze.week1.assignment1.Stock;
import nl.hanze.week1.assignment1.StockGrabber;

/**
 * Created by Laurens on 25-9-2017.
 */
public class AAPLObserver extends BaseObserver implements Observer {

    private double aaplPrice;
    private StockGrabber stockGrabber;

    public AAPLObserver(StockGrabber stockGrabber) {
        stockGrabber.register(this);
        this.stockGrabber = stockGrabber;
    }

    @Override
    public void update(double ibmPrice, double aaplPrice, double googPrice) {
        this.aaplPrice = aaplPrice;
        updateViews(Stock.AAPL, aaplPrice);
    }
}

File: src/main/java/nl/hanze/assignment1/observers/AAPLObserver.kt
package nl.hanze.assignment1.observers

import nl.hanze.assignment1.*

/**
 * Created by Laurens on 25-9-2017.
 */

class AAPLObserver(private val stockGrabber: Subject) : BaseObserver(), Observer {

    private var aaplPrice: Double = 0.0

    init {
        stockGrabber.register(this)
    }

    override fun update(stockUpdate: StockUpdate) {
        if (stockUpdate.stock==Stock.AAPL) {
            aaplPrice = stockUpdate.price
            updateViews(stockUpdate)
        }
    }
}
